lib/salus/scanners/osv/base.rb

```diff
@@ -0,0 +1,133 @@
+require 'salus/scanners/base'
+require 'json'
+require 'zip'
+require 'uri'
+require 'net/http'
+
+module Salus::Scanners::OSV
+  class Base < Salus::Scanners::Base
+    class ApiTooManyRequestsError < StandardError; end
+
+    MAX_RETRIES = 2
+
+    def run
+      raise NoMethodError
+    end
+
+    def should_run?
+      raise NoMethodError
+    end
+
+    private
+
+    def name
+      self.class.name.sub('Salus::Scanners::OSV::', '')
+    end
+
+    def osv_vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities
+    end
+
+    def osv_url
+      # Bucket location contains individual entries in OSVF.
+      url = if @repository.go_sum_present? || @repository.go_mod_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "Go/all.zip"
+            elsif @repository.requirements_txt_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "PyPI/all.zip"
+            elsif @repository.pom_xml_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "Maven/all.zip"
+            elsif @repository.gemfile_present? || @repository.gemfile_lock_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "RubyGems/all.zip"
+            elsif @repository.cargo_present? || @repository.cargo_lock_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "crates.io/all.zip"
+            elsif @repository.package_lock_json_present?
+              "https://osv-vulnerabilities.storage.googleapis.com/"\
+              "npm/all.zip"
+            end
+      URI(url)
+    end
```

Comment:
```
def method_a
    return x if b == 1
    return y if b == 2
    ...
end
```
The above pattern is preferable to if..else in ruby. Search "Prefer guard clause" in https://confluence.coinbase-corp.com/pages/viewpage.action?pageId=823230474

We can write the code as follows:

```
def osv_url
   return osv_url_for("Go") if go_repo?
   return ...  if  ...
   return ...  if  ...
end

private

def go_repo?
  @repository.go_sum_present? || @repository.go_mod_present?
end

def osv_url_for(package)
   url = "https://osv-vulnerabilities.storage.googleapis.com/"
            + package + "all.zip"
   URI(url)
end
```

WDYT? 🤔

Reply:
Yeah that makes sense, let me update that.